src/Microsoft.Identity.Client/ITokenCache.cs

```diff
@@ -64,47 +65,103 @@ public interface ITokenCache
+        /// MSAL desktop versions (MSAL.Python and MSAL.Java for instance).
+        /// If you need to maintain SSO between an application using MSAL 2.x and this application using MSAL 3.x,
+        /// you might also want to serialize with Serialize and Deserialize, otherwise just use
+        /// SerializeMsalV3 and DeserializeMsalV3
         /// </summary>
-        /// <param name="unifiedState"></param>
-        void Deserialize(byte[] unifiedState);
+        /// <param name="bytes">Byte stream representation of the cache</param>
+        /// <param name="merge">Future: Boolean enabling merging the cache into the existing cache.</param>
+        /// <remarks>
+        /// Merge is not yet available.
+        /// </remarks>
+        void DeserializeMsalV3(byte[] bytes, bool merge = false);
+
+        /// <summary>
+        /// Serializes the token cache to the MSAL.NET 3.x unified cache format, which is compatible with other
+        /// MSAL desktop versions (MSAL.Python and MSAL.Java for instance).
+        /// If you need to maintain SSO between an application using MSAL 2.x and this application using MSAL 3.x,
+        /// you might also want to serialize with Serialize and Deserialize, otherwise just use
+        /// SerializeMsalV3 and DeserializeMsalV3.
+        /// </summary>
+        /// <returns>Byte stream representation of the cache</returns>
+        byte[] SerializeMsalV2();
+
+        /// <summary>
+        /// Deserializes the token cache to the MSAL.NET 3.x unified cache format, which is compatible with other
+        /// MSAL desktop versions (MSAL.Python and MSAL.Java for instance).
+        /// If you need to maintain SSO between an application using MSAL 2.x and this application using MSAL 3.x,
+        /// you might also want to serialize with Serialize and Deserialize, otherwise just use
+        /// SerializeMsalV3 and DeserializeMsalV3
+        /// </summary>
+        /// <param name="bytes">Byte stream representation of the cache</param>
+        /// <param name="merge">Future: Boolean enabling merging the cache into the existing cache.</param>
+        /// <remarks>
+        /// Merge is not yet available.
+        /// </remarks>
+        void DeserializeMsalV2(byte[] bytes, bool merge = false);
 
         /// <summary>
         /// Serializes the token cache to the MSAL.NET 3.x unified cache format, which is compatible with other
         /// MSAL desktop versions (MSAL.Python and MSAL.Java for instance).
         /// If you need to maintain SSO between an application using MSAL 2.x and this application using MSAL 3.x,
         /// you might also want to serialize with Serialize and Deserialize, otherwise just use
-        /// SerializeV3 and DeserializeV3.
+        /// SerializeMsalV3 and DeserializeMsalV3.
         /// </summary>
         /// <returns>Byte stream representation of the cache</returns>
-        byte[] SerializeV3();
+        byte[] SerializeAdalV3();
```

Comment:
>SerializeAdalV3 [](start = 15, length = 15)

this says SerializeAdalV3, is the summary right still?

Reply:
please see PR comment. next update will fix all the summaries

Reply:
ok.

---
In reply to: [258297595](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/895#discussion_r258297595) [](ancestors = 258297595)